diff --git a/vendor/github.com/hashicorp/go-discover/discover.go b/vendor/github.com/hashicorp/go-discover/discover.go
index 369fd4153..6d4de6dcb 100644
--- a/vendor/github.com/hashicorp/go-discover/discover.go
+++ b/vendor/github.com/hashicorp/go-discover/discover.go
@@ -9,6 +9,7 @@ import (
 	"strings"
 	"sync"
 
+/*
 	"github.com/hashicorp/go-discover/provider/aliyun"
 	"github.com/hashicorp/go-discover/provider/aws"
 	"github.com/hashicorp/go-discover/provider/azure"
@@ -23,6 +24,7 @@ import (
 	"github.com/hashicorp/go-discover/provider/tencentcloud"
 	"github.com/hashicorp/go-discover/provider/triton"
 	"github.com/hashicorp/go-discover/provider/vsphere"
+*/
 )
 
 // Provider has lookup functions for meta data in a
@@ -45,6 +47,7 @@ type ProviderWithUserAgent interface {
 
 // Providers contains all available providers.
 var Providers = map[string]Provider{
+/*
 	"aliyun":       &aliyun.Provider{},
 	"aws":          &aws.Provider{},
 	"azure":        &azure.Provider{},
@@ -59,6 +62,7 @@ var Providers = map[string]Provider{
 	"triton":       &triton.Provider{},
 	"vsphere":      &vsphere.Provider{},
 	"packet":       &packet.Provider{},
+*/
 }
 
 // Discover looks up metadata in different cloud environments.
