From 628fde253de796edc6f87ee358c2eaee98c7b94a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Anders=20F=20Bj=C3=B6rklund?= <anders.f.bjorklund@gmail.com>
Date: Sat, 9 Apr 2022 13:56:52 +0200
Subject: [PATCH] Restore missing defaults for cni options

---
 src/cmd/cri/options/options.go | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/cmd/cri/options/options.go b/src/cmd/cri/options/options.go
index d02641d..949c68a 100644
--- a/src/cmd/cri/options/options.go
+++ b/src/cmd/cri/options/options.go
@@ -113,5 +113,9 @@ func NewContainerRuntimeOptions() *config.ContainerRuntimeOptions {
 		CriDockerdRootDirectory:   "/var/lib/cri-dockerd",
 		PodSandboxImage:           defaultPodSandboxImage,
 		ImagePullProgressDeadline: metav1.Duration{Duration: 1 * time.Minute},
+
+		CNIBinDir:   "/opt/cni/bin",
+		CNIConfDir:  "/etc/cni/net.d",
+		CNICacheDir: "/var/lib/cni/cache",
 	}
 }
-- 
2.25.1

